packages/astro/src/@types/astro.ts

```diff
@@ -349,14 +349,24 @@ export interface AstroUserConfig {
 		 */
 		drafts?: boolean;
 
+		/**
+		 * @docs
+		 * @name markdown.mode
+		 * @type {'md' | 'mdx'}
+		 * @default `mdx`
+		 * @description
+		 * Control wheater to allow components inside markdown files ('mdx') or not ('md').
+		 */
+		mode?: 'md' | 'mdx';
+
```

Comment:
I don't think this actually does anything right now, which is why we didn't include it!

Reply:
It does add `rehypeJsx` and `rehypeExpressions`. I could remove the MDX mode altogether if these plugins do nothing

Reply:
Ah no you're right! I also saw that https://github.com/withastro/astro/pull/2971 added escaping for expressions when `mode === 'md'`. So I think this is fine!

Reply:
After doing some testing, the only difference I've found is that components are being rendered with `mdx` but completely ignored with `md`. Expressions works in both modes, tho.